sympy/physics/mechanics/joint.py

```diff
@@ -543,3 +543,137 @@ def _set_linear_velocity(self):
+        axis. Default is x axis in parent's frame.
+    child_axis : Vector, optional
+        Axis of child frame which would be aligned with parent's
+        axis. Default is x axis in child's frame.
+
+    Attributes
+    ==========
+
+    name : string
+        The joint's name.
+    parent : Body
+        The joint's parent body.
+    child : Body
+        The joint's child body.
+    coordinate : dynamicsymbol, optional
+        Coordinate of the joint.
+    speed : dynamicsymbol, optional
+        Speed of the joint.
+    parent_joint : Point
+        The joint's point where parent body is connected to the joint.
+    child_joint : Point
+        The joint's point where child body is connected to the joint.
+    parent_axis : Vector
+        The axis of parent frame.
+    child_axis : Vector
+        The axis of child frame.
+
+    Examples
+    =========
+
+    This is an example of linear mass spring damper where we will do all kinematics
+    using SlidingJoints.
+
+    >>> from sympy import symbols
+    >>> from sympy.physics.mechanics import SlidingJoint, Body
+    >>> from sympy.physics.vector import dynamicsymbols, ReferenceFrame
+
+    Declaring the mass, frame, speeds, coordinates etc isn't necessary as `Body` and
+    `SlidingJoint` can declare it themselves. But we would be declaring them for better understanding.
+
+    >>> x, v = dynamicsymbols('x v')
+    >>> m = symbols('m')
+    >>> N = ReferenceFrame('N')
+
+    Declaring the bodies.
+
+    >>> C = Body('C', frame=N)
+    >>> P = Body('P', frame=N, mass=m)
+
+    Bodies are connected through their masscenters.
+
+    >>> J = SlidingJoint('J', C, P, coordinates=x, speeds=v)
+
+    We can check the kinematics now.
+
+    >>> P.masscenter.pos_from(C.masscenter)
+    x(t)*N.x
+    >>> P.masscenter.vel(N)
+    v(t)*N.x
+    >>> J.kdes()
+    [v(t) - Derivative(x(t), t)]
+
+    """
+
+    def __init__(self, name, parent, child, coordinates=None, speeds=None, parent_joint_pos=None,
+        child_joint_pos=None, parent_axis=None, child_axis=None):
+
+        super().__init__(name, parent, child, coordinates, speeds, parent_joint_pos, child_joint_pos,
+            parent_axis, child_axis)
+
+    def _generate_coordinates(self, coordinate):
+        coordinates = []
+        if coordinate is None:
+            x = dynamicsymbols('x' + '_' + self._name)
+            coordinate = x
+        coordinates.append(coordinate)
+        return coordinates
+
+    def _generate_speeds(self, speed):
+        speeds = []
+        if speed is None:
+            y = dynamicsymbols('v' + '_' + self._name)
+            speed = y
+        speeds.append(speed)
+        return speeds
+
+    def _orient_frames(self):
+        self._child.frame.orient_axis(self._parent.frame, self._parent_axis, 0)
+        angle, axis = self._align_axis(self._parent_axis, self._child_axis)
+        if axis != Vector(0):
+            self._child.frame.orient_axis(self._parent.frame, axis, angle)
```

Comment:
I think this will need to be the same as PinJoint's implementation, no? Maybe this method belongs in `Joint`.